hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/tracing/TraceScope.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.tracing;
+
+import java.io.Closeable;
+
+public class TraceScope implements Closeable {
+  Span span;
+
+  public TraceScope(Span span) {
+    this.span = span;
+  }
+
+  // Add tag to the span
+  public Span addKVAnnotation(String key, String value) {
+    return span;
+  }
+
+  public Span addKVAnnotation(String key, Number value) {
+    return span;
+  }
+
+  public Span addTimelineAnnotation(String msg) {
+    return span;
+  }
+
+  public Span span() {
+    return span;
+  }
+
+  public Span getSpan() {
+    return span;
+  }
+
+  public void reattach() {
+  }
+
+  public void detach() {
+  }
+
+  public void close() {
+    span.close();
```

Comment:
any risk of span ever == null?

Reply:
Thanks @steveloughran for taking a look.
Will add an `if` check.